recipes/date/all/test_package/CMakeLists.txt

```diff
@@ -8,4 +8,3 @@ conan_basic_setup()
 
 add_executable(${PROJECT_NAME} test_package.cpp)
 target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
-set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```

Comment:
I see C++11 features in date public headers. This line shouldn't be removed if you want to support gcc 4.9 (and maybe old clang versions with C++11 support, not sure).

Reply:
my understanding is that in order for clients to use this library with gcc 4.9 they need to use compiler.cppstd=11 flag.
Then that flag would propagate downwards to test_package as well.
Otherwise this flag was giving false positives

Reply:
I would sayt that test_package is more for CCI than package user. If you remove this line, CCI can't succeed to compile test_package with several compilers versions eventhough it could.
Also, I've never seen `tools.check_min_cppstd` used for C++11 libraries. If you have only 82 packages generated, I think that this check is too strict.

Reply:
If you see an error in my logic - please do reopen this conversation.

Reply:
I don' have permission to reopen this conversation, please reopen it ;)

Reply:
@SpaceIm  Reopened.
More comments from me:
Please do take a note that in the conanfile i've now added the 'mincppstd' check. the library is compatible only with c++11 and ownards. 
How it will work in my understanding is that test_package will:
When compiler.cppstd(setting) is defined it will take that compilation flag and use it when building the projects. 
When cppstd is not defined it will use the compilers default.

Adding this line - forces test_package to build with c++11 always - which shouldnt be the case - it should build in the same cppstd as the package itself.

Reply:
cppstd is experimental: https://docs.conan.io/en/latest/howtos/manage_cpp_standard.html

I suggest to:
1/ remove the first part of `cmake.patch`, let CMake decides to use the leatest C++ standard up to C++17, otherwise compilers without at least C++11 as default standard will fail to compile (without cppstd in conan profile, but cppstd is not set AFAIK in CI of conan center index).
2/ remove `tools.check_min_cppstd` in configure of conanfile
3/ add `set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)` in CMakeLists of test_package (same reason than 1/ )

With these 3 modifications, you will be able to build date for all supported triplet of CCI (I've tested just few minutes ago: https://github.com/SpaceIm/conan-date)

Reply:
I am against that change.
its library headers are not compile without c++11. 
even though its experimental (cppstd) they are recipes here in the conan-center-index that use it.
and its not like its behind some kind of feature gate. 

anyway, because the original cmake  does this magic:
https://github.com/HowardHinnant/date/blob/v2.4.1/CMakeLists.txt#L10
which actually is something i've removed - because i am believer of cppstd, i will bring this back and add this set_property as well.

Reply:
You can keep `check_min_cppstd` if you want, but please follow what is done most recipes using it:
```python
    def configure(self):
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, 11)
```

Reply:
applied.

Reply:
@SpaceIm  slightly grumpy "okay, thats good enough" :) i'll add the  configure back in. with the if.

Reply:
@SpaceIm  Actually now that you mentioned this 'if' only then i saw the errors of my ways.
so thanks for that last comment.

Basically there is no sense to do check_min_cppstd forcefully without if statement on this general repository, as this recipe may be used in a vast different of ways, 
some of them will set their cppstd by the same means as original cmake does.